apps/backend/Dockerfile

```diff
@@ -0,0 +1,36 @@
+FROM node:14 as base
+
+RUN curl -f https://get.pnpm.io/v6.16.js | node - add --global pnpm
+
+WORKDIR /app
+
+# pnpm fetch does require only lockfile
+COPY pnpm-lock.yaml .
+RUN pnpm fetch
+
+ADD . .
```

Comment:
To avoid unexpected results you could use COPY here.

https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#:~:text=ADD%20or%20COPY&text=COPY%20only%20supports%20the%20basic,rootfs.tar.xz%20%2F%20.